samples/nrf9160/serial_lte_modem/src/slm_at_icmp.c

```diff
@@ -0,0 +1,496 @@
+		if (data[0] != rep) {
+			/* We got unsolicited message
+			 * drop it and wait for echo-reply
+			 */
+			goto reread;
+		} else {
+			if (ip_buf[IP_PROTOCOL_POS] != ICMP) {
+				/* Not ipv4 echo reply, ignore silently */
+				goto reread;
+			} else {
+				/* Check ICMP HCS */
+				hcs = check_ics(data, len - header_len);
+				if (hcs != 0) {
+					LOG_ERR("HCS error %d", hcs);
+					goto close_end;
+				}
+				/* Payload length */
+				pllen = (ip_buf[2] << 8) + ip_buf[3];
+			}
+
+			/* Check seqnr and length */
+			plseqnr = data[7];
+			if (plseqnr != seqnr) {
+				LOG_ERR("error sequence numbers %d %d",
+					plseqnr, seqnr);
+				goto close_end;
+			}
+			if (pllen != len) {
+				LOG_ERR("error length %d %d", pllen, len);
+				goto close_end;
+			}
+
+			/* Result */
+			sprintf(buf, "#XPING: %d.%03d\r\n",
+				(u32_t)(delta_t)/1000,
+				(u32_t)(delta_t)%1000);
+			m_callback(buf);
+		}
+
+close_end:
+		(void)nrf_close(fd);
+	} while (0);
+
+	return (u32_t)delta_t;
+}
+
+void ping_task(struct k_work *item)
+{
+	struct addrinfo *si = ping_argv.src;
+	struct addrinfo *di = ping_argv.dest;
+	u32_t sum = 0;
+	u32_t count = 0;
+
+	ARG_UNUSED(item);
+
+	for (int i = 0; i < ping_argv.count; i++) {
+		u32_t to = send_ping_wait_reply();
+
+		if (to > 0)  {
+			count++;
+			sum += to;
+		}
+		k_sleep(ping_argv.interval);
+	}
+
+	if (count > 1) {
+		u32_t ave = (sum + count/2) / count;
+		int ave_s = ave / 1000;
+		int ave_f = ave % 1000;
+
+		sprintf(buf, "#XPING: average %d.%03d\r\n", ave_s, ave_f);
+		m_callback(buf);
+		m_callback("OK\r\n");
+	}
+
+	freeaddrinfo(si);
+	freeaddrinfo(di);
+}
+
+static int ping_test_handler(const char *url, int length, int waittime,
+			int count, int interval)
+{
+	int st;
+	struct addrinfo *res;
+	int addr_len;
+
+	st = modem_info_params_get(&modem_param);
+	if (st < 0) {
+		LOG_ERR("Unable to obtain modem parameters (%d)", st);
+		return -1;
+	}
+
+	/* Check network connection status by checking local IP address */
+	addr_len = strlen(modem_param.network.ip_address.value_string);
+	if (addr_len == 0) {
+		LOG_ERR("LTE not connected yet");
+		return -1;
+	}
+	st = getaddrinfo(modem_param.network.ip_address.value_string,
+			NULL, NULL, &res);
+	if (st != 0) {
+		LOG_ERR("getaddrinfo(src) error: %d", st);
+		return -st;
+	}
+	ping_argv.src = res;
+
+	/* Get destination */
+	res = NULL;
+	st = getaddrinfo(url, NULL, NULL, &res);
+	if (st != 0) {
+		LOG_ERR("getaddrinfo(dest) error: %d", st);
+		freeaddrinfo(ping_argv.src);
+		return -st;
+	}
+	ping_argv.dest = res;
+
+	if (ping_argv.src->ai_family != ping_argv.dest->ai_family) {
+		LOG_ERR("Source/Destination address family error");
+		freeaddrinfo(ping_argv.dest);
+		freeaddrinfo(ping_argv.src);
+		return -1;
+	}
+
+	ping_argv.len = length;
+	ping_argv.waitms = waittime;
+	ping_argv.count = 1;		/* default 1 */
+	ping_argv.interval = 1000;	/* default 1s */
+	if (count > 0) {
+		ping_argv.count = count;
+	}
+	if (interval > 0) {
+		ping_argv.interval = count;
```

Comment:
Is it intentional to assign 'count' and not 'interval' here?

Reply:
This is a bug, thanks